src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -33,6 +36,32 @@ public static boolean isValidTagName(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Set the tag category.
+     * @param tagCategory
+     */
+    public void setTagCategory(TagCategory tagCategory) {
```

Comment:
Should tagCategory be passed as String or TagCategory object?

Reply:
I prefer TagCategory object

Reply:
I would feel TagCategory Object would  be a better practice for abstraction.